api/kubernetes/cli/service_account.go

```diff
@@ -53,7 +40,18 @@ func (kcl *KubeClient) ensureServiceAccountForUserExists(serviceAccountName stri
 		return err
 	}
 
-	return kcl.ensureNamespaceAccessForServiceAccount(serviceAccountName, defaultNamespace)
+	if !restrictDefaultNamespace {
+		err = kcl.ensureNamespaceAccessForServiceAccount(serviceAccountName, defaultNamespace)
+		if err != nil {
+			return err
+		}
+	}
```

Comment:
Maybe it's a backport and we don't want to change patterns too much, but do you think we could move this to the `setupNamespaceAccesses()`?
I was thinking about changing the for-loop body (line https://github.com/portainer/portainer/pull/5337/files#diff-b3fdf6d37860338fbd688672b9f6ce31ea467f9e1ef3c46fb2deaa81919243b7R58) to that
```
policies, ok := accessPolicies[namespace.Name]
if !ok || !hasUserAccessToNamespace(userID, teamIDs, policies) {
	err = kcl.removeNamespaceAccessForServiceAccount(serviceAccountName, namespace.Name)
	if err != nil {
		return err
	}
	continue
}

if namespace.Name == defaultNamespace && !restrictDefaultNamespace {
	err = kcl.ensureNamespaceAccessForServiceAccount(serviceAccountName, namespace.Name)
	if err != nil && !k8serrors.IsAlreadyExists(err) {
		return err
	}
}
```

Not only that would optimise conditions by squashing `!ok` and `!hasUserAccessToNamespace`, but also ensure that a user don't have "dangling" resources in the default namespace, as well as encapsulating the rolebindings manipulations in the same method and having less of `!restrictDefaultNamespace` checks.